src/Compiler.php

```diff
@@ -398,17 +399,49 @@ public function setErrorOuput($handle)
     /**
      * Compile scss
      *
-     * @api
+     * @param string      $code
+     * @param string|null $path
      *
-     * @param string $code
-     * @param string $path
+     * @return string
      *
-     * @return CompilationResult
+     * @throws SassException when the source fails to compile
+     *
+     * @deprecated Use {@see compileString} instead.
      */
     public function compile($code, $path = null)
+    {
+        @trigger_error(sprintf('The "%s" method is deprecated. Use "compileString" instead.', __METHOD__), E_USER_DEPRECATED);
+
+        $result = $this->compileString($code, $path);
+
+        $sourceMap = $result->getSourceMap();
+
+        if ($sourceMap !== null) {
+            if ($this->sourceMap instanceof SourceMapGenerator) {
+                $this->sourceMap->saveMap($sourceMap);
+            } elseif ($this->sourceMap === self::SOURCE_MAP_FILE) {
+                $sourceMapGenerator = new SourceMapGenerator($this->sourceMapOptions);
+                $sourceMapGenerator->saveMap($sourceMap);
+            }
+        }
+
+        return $result->getCss();
+    }
+
+    /**
+     * Compile scss
+     *
+     * @param string      $source
+     * @param string|null $path
+     *
+     * @return CompilationResult
+     *
+     * @throws SassException when the source fails to compile
+     */
+    public function compileString($source, $path = null)
```

Comment:
I named the parameter `source` rather than `code`, as I think that's a better name for what it is. However, note that I'm not considering parameter names as covered by backward compatibility yet. I haven't reviewed the full codebase regarding names. So using PHP 8 named parameters is still something downstream projects will do at their own risk.